.tekton/.currency/scripts/generate_report.py

```diff
@@ -0,0 +1,184 @@
+        page = requests.get(url)
+        soup = BeautifulSoup(page.text, "html.parser")
+        # ASGI
+        if "asgi" in url:
+            text = (
+                soup.find(id="version-history")
+                .findChild("li", string=re.compile(pattern))
+                .text
+            )
+        # WSGI
+        else:
+            tag = soup.find(id="numerical-index").find_all(
+                "a", string=re.compile("Web Server Gateway Interface")
+            )[-1]
+            text = tag.text
+        res = re.search(pattern, text)
+        return res[1]
+
+    else:
+        # get info using PYPI API
+        response = requests.get(f"{PIP_INDEX_URL}/{dependency}/json")
+        response_json = response.json()
+        latest_version = response_json["info"]["version"]
+        return latest_version
+
+
+def get_last_supported_version(tekton_ci_output, dependency):
+    """get up-to-date supported version"""
+    pattern = r"-([^\s]+)"
+
+    if dependency == "Psycopg2":
+        dependency = "psycopg2-binary"
+
+    last_supported_version = re.search(
+        dependency + pattern, tekton_ci_output, flags=re.I | re.M
+    )
+
+    return last_supported_version[1]
+
+
+def isUptodate(last_supported_version, latest_version):
+    if last_supported_version == latest_version:
+        up_to_date = "Yes"
+    else:
+        up_to_date = "No"
+
+    return up_to_date
+
+
+def get_taskruns(namespace, task):
+    group = "tekton.dev"
+    version = "v1"
+    plural = "taskruns"
+
+    # access the custom resource from tekton
+    tektonV1 = client.CustomObjectsApi()
+    taskruns = tektonV1.list_namespaced_custom_object(
+        group,
+        version,
+        namespace,
+        plural,
+        label_selector=f"{group}/task={task}, triggers.tekton.dev/trigger=python-tracer-scheduled-pipeline-triggger",
+    )["items"]
+
+    taskruns.sort(key=lambda tr: tr["metadata"]["creationTimestamp"], reverse=True)
+
+    return taskruns
+
+
+def get_tekton_ci_output():
+    # config.load_kube_config()
+    config.load_incluster_config()
+
+    namespace = "default"
+
+    starlette_taskruns = get_taskruns(
+        namespace, task="python-tracer-unittest-gevent-starlette-task"
+    )
+
+    coreV1 = client.CoreV1Api()
+    tekton_ci_output = ""
+    for tr in starlette_taskruns:
+        if tr["status"]["conditions"][0]["type"] == "Succeeded":
+            pod = tr["status"]["podName"]
+            logs = coreV1.read_namespaced_pod_log(
+                pod, namespace, container="step-unittest"
+            )
+            if "Successfully installed" in logs:
+                match = re.search("Successfully installed .* (starlette-[^\s]+)", logs)
+                tekton_ci_output += f"{match[1]}\n"
+                break
+
+    default_taskruns = get_taskruns(
+        namespace, task="python-tracer-unittest-default-task"
+    )
+
+    for tr in default_taskruns:
+        if (
+            tr["metadata"]["name"].endswith("unittest-default-3")
```

Comment:
Could we put these conditions into a lambda and pass it as a `filter` argument to `get_taskruns`?